test/cart-payload-mapper-test.js

```diff
@@ -471,15 +471,45 @@ suite('cart-payload-mapper-test', () => {
   });
 
   suite('webUrl', () => {
-    test('it returns checkout URL', () => {
+    test('it returns checkout URL, with _fd=0 appended if not present', () => {
       const cart = {
         checkoutUrl: 'https://shop.com/checkout'
       };
 
       const result = mapCartPayload(cart);
 
+      assert.strictEqual(result.webUrl, `${cart.checkoutUrl}?_fd=0`);
+    });
+
+    test('it gracefully handles invalid URLs by returning them unchanged', () => {
+      const cart = {
+        checkoutUrl: 'invalid'
+      };
+
+      const result = mapCartPayload(cart);
+
+      assert.strictEqual(result.webUrl, cart.checkoutUrl);
+    });
+
+    test('it does not duplicate the query param _fd=0 if it is already present', () => {
+      const cart = {
+        checkoutUrl: 'https://shop.com/checkout?_fd=0'
+      };
+
+      const result = mapCartPayload(cart);
+
       assert.strictEqual(result.webUrl, cart.checkoutUrl);
     });
+
+    test('it updates the value of _fd to 0 if it is present but with a different value', () => {
```

Comment:
Looks great, I'd a two more tests!

1. the url has already other params in it (without `_fd`)
2. the url has already other params and `_fd` too